build_katran_centos.sh

```diff
@@ -0,0 +1,547 @@
+#!/usr/bin/env bash
+
+ # Copyright (C) 2018-present, Facebook, Inc.
+ #
+ # This program is free software; you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation; version 2 of the License.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License along
+ # with this program; if not, write to the Free Software Foundation, Inc.,
+ # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+set -xeo pipefail
```

Comment:
Copy pasting a script N times for each variation of build is not going to be sustainable. Can we extract the platform specific logic into separate scripts, and then have the main script (build_katran.sh) call it based on the platform?

Reply:
@udippant fair point... I could move build_katran_ubuntu.sh back to build_katran.sh and then change the individual methods to have if centos do this, else do the ubuntu logic, only adding it where the logic differs. Would that be acceptable?

Reply:
@ryanohnemus that would be better, yes.
Alternatively, you can also extract out the 'if centos' part to one 'module' and 'else ubuntu' part to another module.
So in the build_katran.sh, you just initialize the platform specific module once in the beginning and delegate when necessary.